### PR TITLE
Add option to define custom render pass file

### PR DESCRIPTION
**Pull request by axb2035**
This allows allow a user to define a custom render pass file. This means that user can create their own render pass files instead of editing example.renderpasses.ini, the same way you can define a custom config.ini file.

Tested all four combos - UseExampleRenderPasses yes/no; RenderPassFile commented out/uncommented.

If UseExampleRenderPasses = yes then RenderPassFile ignored. If UseExampleRenderPasses = no and RenderPassFile commented out main() drops through without doing anything. 

Also, added check that file exists. If the render pass file cannot be found there is a graceful exit with message.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -42,6 +42,12 @@ def main():
 
 	if defaults.getboolean("UseExampleRenderPasses"):
 		config.read_string(fix_config("config/example.renderpasses.ini"))
+	elif defaults.get("RenderPassFile"):		
+		if os.path.isfile(defaults.get("RenderPassFile")):
```
Comment (zamp) [inline main.py:46]: Store the RenderPassFile into a variable instead of getting it again from config

File: main.py [hunk 2]
```diff
@@ -42,6 +42,12 @@ def main():
...
+			config.read_string(fix_config(defaults.get("RenderPassFile")))
```
Comment (zamp) [inline main.py:47]: Same here, use a variable.

File: main.py [hunk 3]
```diff
@@ -42,6 +42,12 @@ def main():
...
+		else:
+			print(f"Could not find RenderPassFile: {defaults.get('RenderPassFile')}")
```
Comment (zamp) [inline main.py:49]: Yet again 👍

File: .gitignore [hunk 4]
```diff
@@ -5,3 +5,7 @@
 /config.py
 /config.ini
 /config/config.ini
+/python_embeded
+/sequences
+/workflows/sequences
```
Comment (zamp) [inline .gitignore:10]: Is this ignore needed or did you commit this by mistake?

**Review by zamp — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (axb2035) [reply main.py:46]: Done.

File: main.py (hunk 2 above)
Comment (axb2035) [reply main.py:47]: Done.

File: main.py (hunk 3 above)
Comment (axb2035) [reply main.py:49]: And done again :)

File: .gitignore (hunk 4 above)
Comment (axb2035) [reply .gitignore:10]: This comes down to my inexperiecne with PRs. I had everything set up in one location and I committed this to stop it trying to add thousands of files. When I built the PR it autmatically added the commit and I can't seem to find a way to take it out or revert it because more commits had been made. This morning I deleted the extra lines and recommitted so it should be in line with the current `main`.

I've set up a repo folder and a test folder elsewhere with all the extra stuff (python, videos, etc), so this won't be a problem going forward.

**Comment by axb2035** [comment]
FYI: I'm going to try to cerate a new branch on my vid2vid fork and see if I can then get two PR going at the same time. If it messes up, is suspect it will end up in this PR.

File: main.py [hunk 5]
```diff
@@ -39,9 +39,16 @@ def main():
 	config.read_string(fix_config("config/config.ini"))
 
 	defaults = config["DEFAULT"]
-
+	render_pass_file = defaults.get("RenderPassFile")
+								 
 	if defaults.getboolean("UseExampleRenderPasses"):
 		config.read_string(fix_config("config/example.renderpasses.ini"))
+	elif render_pass_file:
```
Comment (zamp) [inline main.py:46]: This condition seems suspicious.
Just an "else" is fine in this case.

**Review by zamp — CHANGES_REQUESTED** [review]

File: main.py (hunk 5 above)
Comment (axb2035) [reply main.py:46]: If RenderPassFile is not defined in one of the configs then `render_pass_file == None`. Using an `else` causes an error a few lines later if UseExampleRenderPasses is also set to `no` or commented out when it tests to see if the file exists.

Agree, it is a bit of a corner case, but would rather err on the side of trapping errors and giving meaningful feedback than letting it bomb.

Comment (zamp) [reply main.py:46]: It's fine to print out an error in this case since if the user doesn't use example render passes then they should have their own. Now it will run the script without any render passes and it won't print any error about not having any which is really weird behavior for the user.

Comment (axb2035) [reply main.py:46]: Before this PR, if a user has no renderpasses in either the examplerenderpasses or config files, then it drops through just printing DONE!  So, this PR introduces no new behaviour. Agree that itbe better that zero renderpasses are flagged in all situations.

**Comment by axb2035** [comment]
Added a check after config has loaded to see if there are any render passes. Graceful exit with message if none identified.

**Review by zamp — APPROVED** [review]
